pkg/analyzer/analyze.go

```diff
@@ -0,0 +1,665 @@
+    os.Exit(0)
+  }
+
+  if uri && purl  || uri && id || purl && id {
+    fmt.Println("Must provide only one of --uri or --purl")
+    os.Exit(0)
+  }
+}
+
+
+func GenerateAnalysisOutput(analysisGraph graph.Graph[string, *Node], diffList HighlightedDiff, all, dot bool, maxprint, action int, gqlclient graphql.Client){
+  //Create dot file
+  createGraphDotFile(dot, analysisGraph)
+  //print to stdout
+  printHighlightedAnalysis(dot, diffList, all, maxprint, action, analysisGraph )
+}
+func max(nums []int) int {
+  if len(nums) == 0 {
+    return 0
+  }
+  max := nums[0]
+  for _, num := range nums[1:] {
+    if num > max {
+      max = num
+    }
+  }
+  return max
+}
+func printHighlightedAnalysis(dot bool,diffList HighlightedDiff, all bool, maxprint, action int,  analysisGraph graph.Graph[string, *Node]){
+
+  if dot {
+    return 
+  }
+  //use action here to do different things
+  if action == 0 {
+    metadataTable := tablewriter.NewWriter(os.Stdout)
+    metadataTable.SetHeader([]string{ "Metadata"})
+    for _, metadata := range diffList.MetadataMismatch {
+      if (!all && len(diffList.MetadataMismatch) == maxprint){
+        break
+      }
+      metadataTable.Append([]string{metadata})
+    }
+    metadataTable.SetAlignment(tablewriter.ALIGN_LEFT)
+    metadataTable.Render()
+  }
+
+
+  table := tablewriter.NewWriter(os.Stdout)
+
+  switch action { 
+  case 0:
+    table.SetHeader([]string{ "Missing Nodes", "Missing Links"})
+  case 1:
+    table.SetHeader([]string{ "Common Nodes", "Common Links"})
+  case 2:
+    table.SetHeader([]string{ "Added Nodes", "Added Links"})
+  }
+
+
+  max :=  max([]int{len(diffList.MissingAddedRemovedNodes), len(diffList.MissingAddedRemovedLinks)})
+
+  for i :=0 ; i< max; i++{
+
+    if (!all && i+1 == maxprint){
+      break
+    }
+
+    var appendList []string
+
+
+    if i< len(diffList.MissingAddedRemovedNodes){
+      namespace, ok := GetNodeAttribute(analysisGraph,diffList.MissingAddedRemovedNodes[i], "Namespace[0]").(string)
+
+      if !ok {
+        fmt.Println("Error getting node namespace attribute")
+        os.Exit(1)
+      }
+      appendList = append(appendList,namespace)
+    } else {
+      appendList = append(appendList, "")
+    }
+
+    if i< len(diffList.MissingAddedRemovedLinks){
+      namespaceOne, okOne := GetNodeAttribute(analysisGraph,diffList.MissingAddedRemovedLinks[i][0], "Namespace[0]").(string)
+      namespaceTwo, okTwo := GetNodeAttribute(analysisGraph,diffList.MissingAddedRemovedLinks[i][1], "Namespace[0]").(string)
+
+      if !okOne || !okTwo {
+        fmt.Println("Error getting node namespace attribute")
+        os.Exit(1)
+      }
+
+      appendList = append(appendList, namespaceOne + "--->"+namespaceTwo)
+    }else {
+      appendList = append(appendList, "")
+    }
+    table.Append(appendList)
+  }
+
+
+
+  table.SetAlignment(tablewriter.ALIGN_LEFT)
+  table.Render()
+  if (!all && max > maxprint){
+    fmt.Println("Run with --all to see full list")
```

Comment:
should user logger here: https://github.com/guacsec/guac/blob/main/pkg/handler/processor/process/process.go#L117

It should be able to be pulled from the context:

```
logger := logging.FromContext(ctx)
```